src/smt/solver_engine.cpp

```diff
@@ -1594,37 +1596,23 @@ void SolverEngine::printInstantiations(std::ostream& out)
   // First, extract and print the skolemizations
   bool printed = false;
   bool reqNames = !d_env->getOptions().quantifiers.printInstFull;
-  // only print when in list mode
-  if (d_env->getOptions().quantifiers.printInstMode == options::PrintInstMode::LIST)
-  {
-    std::map<Node, std::vector<Node>> sks;
-    qe->getSkolemTermVectors(sks);
-    for (const std::pair<const Node, std::vector<Node>>& s : sks)
-    {
-      Node name;
-      if (!qe->getNameForQuant(s.first, name, reqNames))
-      {
-        // did not have a name and we are only printing formulas with names
-        continue;
-      }
-      SkolemList slist(name, s.second);
-      out << slist;
-      printed = true;
-    }
-  }
 
-  // Second, extract and print the instantiations
+  // Extract the skolemizations and instantiations
+  std::map<Node, std::vector<Node>> sks;
   std::map<Node, InstantiationList> rinsts;
   if ((d_env->getOptions().smt.produceProofs
        && d_env->getOptions().smt.proofMode == options::ProofMode::FULL)
       && getSmtMode() == SmtMode::UNSAT)
   {
-    // minimize instantiations based on proof manager
-    getRelevantInstantiationTermVectors(
-        rinsts, options().driver.dumpInstantiationsDebug);
+    // minimize skolemizations and instantiations based on proof manager
+    getRelevantQuantTermVectors(
+        rinsts, sks, options().driver.dumpInstantiationsDebug);
   }
   else
   {
+    // get all skolem term vectors
+    qe->getSkolemTermVectors(sks);
+    // get all instantiations
     std::map<Node, std::vector<std::vector<Node>>> insts;
     getInstantiationTermVectors(insts);
```

Comment:
```suggestion
    qe->getInstantiationTermVectors(insts);
```
Safe to do this here since you already have the quantifiers engine. It's odd to use the solver engine version for this and directly the qe version above for the skolems.

Reply:
Right, good catch.